### PR TITLE
feat: implementation of cigar truncation methods ported from fgbio

### PR DESCRIPTION
**Pull request by ameynert**
Ported `truncate`, `truncateToQueryLength`, and `truncateToTargetLength` and related unit tests from [fgbio](https://github.com/fulcrumgenomics/fgbio/blob/7ee60c361f06c4a082a365f707608238383a2deb/src/main/scala/com/fulcrumgenomics/alignment/Alignment.scala#L249)

### CODE REVIEW

File: tests/fgpyo/sam/test_cigar.py [hunk 1]
```diff
@@ -80,3 +80,64 @@ def test_query_alignment_offsets_reversed(
...
+        # Actual truncation to length 50
+        ("60M", 50, "50M"),
+        ("10H50M", 50, "10H50M"),  # Hard clips preserved
+        ("25M10I25M", 50, "25M10I15M"),  # Insertions consume query
+        ("25M10D25M", 50, "25M10D25M"),  # Deletions don't consume query
+        ("50M10S", 50, "50M"),  # Trailing soft clips removed
+        # Additional edge cases
+        ("10M", 0, "*"),  # Truncate to zero
+        ("*", 5, "*"),  # Empty CIGAR
+    ],
```
Comment (nh13) [inline tests/fgpyo/sam/test_cigar.py]: The fgbio-ported cases are a good start. Consider expanding coverage with:

- **length=0 + leading non-consuming elements** — this catches the `_truncate` bug above:
  `("5H10M", 0, "*")`, `("10S10M", 0, "*")` (for target truncation)
- **Non-consuming ops interspersed** — `"10M5D10I5N10M"` at various lengths, verifying D/N pass through for query and I/S pass through for target
- **Element splitting** — `("100M", 1, "1M")`, `("100M", 50, "50M")`, `("100M", 99, "99M")`
- **All operator types** — `=`, `X`, `P` in addition to M/I/D/N/S/H
- **Reverse-compose truncation from the right** — `reversed().truncate_to_query_length(n).reversed()`
- **Length invariant sweep** — for each cigar, assert `truncated.length_on_query() == min(length, original)` for all lengths 0..N

fgbio's tests for reference:
- [Truncation + coalesce tests (lines 51-75)](https://github.com/fulcrumgenomics/fgbio/blob/0c75163/src/test/scala/com/fulcrumgenomics/alignment/AlignmentTest.scala#L51-L75)
- [Implementation (lines 192-293)](https://github.com/fulcrumgenomics/fgbio/blob/0c75163/src/main/scala/com/fulcrumgenomics/alignment/Alignment.scala#L192-L293)

Also consider adding `coalesce()` ([fgbio's `Cigar.coalesce`](https://github.com/fulcrumgenomics/fgbio/blob/0c75163/src/main/scala/com/fulcrumgenomics/alignment/Alignment.scala#L277-L293)) — merges adjacent same-operator elements, useful after truncation or other cigar manipulation.

**Comment by ameynert** [comment]
@nh13 Should the truncation functions throw an error if the requested truncation length is longer than what is available for either query or target?

File: tests/fgpyo/sam/test_cigar.py (hunk 1 above)
Comment (ameynert) [reply tests/fgpyo/sam/test_cigar.py]: #274 opened for porting `coalesce` - with the additional tests this PR is long enough!

**Review by nh13 — APPROVED** [review]
Looks great — all comments addressed, tests are thorough, and the countdown pattern is clean. To answer your question: no, truncation shouldn't throw when the requested length exceeds the cigar's length. fgbio doesn't either — its `while (pos <= len && iter.hasNext)` just returns the full cigar unchanged. It's a "truncate to *at most* N" operation, like Python's `string[:n]` which silently returns the whole string when `n > len(string)`. Your existing `("75M", 100, "75M")` cases already cover this.

One small addition below: a test for the negative-length `ValueError` path you added.
